src/to_curl.rs

```diff
@@ -289,6 +289,21 @@ pub fn translate(args: Cli) -> Result<Command> {
         cmd.arg(interface);
     };
 
+    if !args.resolve.is_empty() {
+        let port = match url.port() {
+            Some(port) => port,
+            None if url.scheme() == "http" => 80,
+            None if url.scheme() == "https" => 443,
+            None => return Err(anyhow!("Unsupported URL scheme: '{}'", url.scheme())),
+        };
```

Comment:
```suggestion
        let port = url
            .port_or_known_default()
            .with_context(|| format!("Unsupported URL scheme: '{}'", url.scheme()))?;
```
(plus an `anyhow::Context` import)